### ISSUE TITLE
The results are way better than the ones reported in the paper.

### ISSUE THREAD

**Issue by rpfly3** [garrickbrazil/M3D-RPN#16]
Hi Garrick, I run your code on KITTI val1 and val2 dataset. However, the results are way better than the ones reported in the paper, especially for Car. Do you have any similar results?
Val1:
![val1](https://user-images.githubusercontent.com/6251482/66680809-36929800-ec26-11e9-9a9c-7cce66d70afa.png)
Val2:
![val2](https://user-images.githubusercontent.com/6251482/66680823-3b574c00-ec26-11e9-8a44-1d59e0d9cbba.png)
Reported in the paper:
![results_reported](https://user-images.githubusercontent.com/6251482/66680928-7c4f6080-ec26-11e9-814d-8d6a13838dc6.png)

**Comment by qaazii** [comment]
> Hi Garrick, I run your code on KITTI val1 and val2 dataset. However, the results are way better than the ones reported in the paper, especially for Car. Do you have any similar results?
> Val1:
> ![val1](https://user-images.githubusercontent.com/6251482/66680809-36929800-ec26-11e9-9a9c-7cce66d70afa.png)
> Val2:
> ![val2](https://user-images.githubusercontent.com/6251482/66680823-3b574c00-ec26-11e9-8a44-1d59e0d9cbba.png)
> Reported in the paper:
> ![results_reported](https://user-images.githubusercontent.com/6251482/66680928-7c4f6080-ec26-11e9-814d-8d6a13838dc6.png)

Hi How do you get results did you trained he network or you just tested the pre-trained weights. 

I trained the network many times and i get the folowing results. i still dont know what is val1 and val2 but i used spllit1. here is the results.. 

densenet121 depth aware ign .5 bg .5 

test_iter 50000 2d car --> easy: 0.8795, mod: 0.8188, hard: 0.6650
test_iter 50000 gr car --> easy: 0.2070, mod: 0.1663, hard: 0.1491
test_iter 50000 3d car --> easy: 0.1497, mod: 0.1258, hard: 0.1016
test_iter 50000 2d pedestrian --> easy: 0.7394, mod: 0.5943, hard: 0.5101
test_iter 50000 gr pedestrian --> easy: 0.0437, mod: 0.0406, hard: 0.0343
test_iter 50000 3d pedestrian --> easy: 0.0354, mod: 0.0296, hard: 0.0298
test_iter 50000 2d cyclist --> easy: 0.6489, mod: 0.4098, hard: 0.4022
test_iter 50000 gr cyclist --> easy: 0.0181, mod: 0.0114, hard: 0.0114
test_iter 50000 3d cyclist --> easy: 0.0187, mod: 0.0114, hard: 0.0114

**Comment by rpfly3** [comment]
Hi @qazi876 , I used the pre-trained weights downed from the repo. Your results seem more reasonable. Did you try to use the pre-trained weights? If so, can you share your results?

BTW, for the val1 and val2, I believe they are the validation sets in the two splits, namely split1/val and split2/val.

**Comment by qaazii** [comment]
yes. you are right about val2/val2. testing pre trained will always give same results... when i tested the pre-trained weights it gives the same values like yours. but i dont know how the author generated it. 

also i dont know how to change the IOU >= .7 OR 0.5 

IF you have any idea Please let me know.. 

Thanks..

**Comment by brazgardian** [comment]
My feeling is that this is the train -> train accuracy, which looking at the results suggests a bit of over fitting compared to the validation. Nevertheless, I am guessing getting that high of accuracy is caused by using the validation images in the training. 

It would be too good to be true if that is the actual performance. 

Regarding the IoU evaluation criteria. I did not implement a parameter for this in the code. Thus, you will need to manually edit the line in the kitti_spit1/devkit/cpp/evaluate_object.cpp file. I believe the overlap criteria is defined around line 59

**Comment by rpfly3** [comment]
@garrickbrazil Thanks for your remind. I split the dataset with the script provided in your repo. I am sure why it has different results between pre-trained model and re-trained model. Do you have any clue?

**Comment by qaazii** [comment]
> @garrickbrazil Thanks for your remind. I split the dataset with the script provided in your repo. I am sure why it has different results between pre-trained model and re-trained model. Do you have any clue?

I think Mr. Garrick Brazil already explained that he is guessing maybe it is caused by using validation images in training..
i dont think if there is any specific reason. anyhow explaining how he split the kitti split1 and kitti split2 may  help.

**Comment by opjang5** [comment]
@garrickbrazil Hi Mr. Garrick Brazil,I got the different result with your pre-trained depth-aware model.I followed your readme.md.First I trained the warmup network without depth-aware module in 50k iterations.Then I used the warmup network as pre-trained model to train network with depth-aware module in 50k iterations.
How can I get the same result as your pre-trained depth-aware model.
This is my result.
[INFO]: 2020-03-01 17:54:55,341 test_iter 50000 2d car --> easy: 0.8925, mod: 0.8331, hard: 0.6742
[INFO]: 2020-03-01 17:54:55,342 test_iter 50000 gr car --> easy: 0.2565, mod: 0.2104, hard: 0.1782
[INFO]: 2020-03-01 17:54:55,342 test_iter 50000 3d car --> easy: 0.1987, mod: 0.1655, hard: 0.1344
[INFO]: 2020-03-01 17:54:55,343 test_iter 50000 2d pedestrian --> easy: 0.6555, mod: 0.5793, hard: 0.4975
[INFO]: 2020-03-01 17:54:55,344 test_iter 50000 gr pedestrian --> easy: 0.0862, mod: 0.0718, hard: 0.0659
[INFO]: 2020-03-01 17:54:55,344 test_iter 50000 3d pedestrian --> easy: 0.0713, mod: 0.0660, hard: 0.0582
[INFO]: 2020-03-01 17:54:55,345 test_iter 50000 2d cyclist --> easy: 0.6476, mod: 0.4077, hard: 0.4054
[INFO]: 2020-03-01 17:54:55,345 test_iter 50000 gr cyclist --> easy: 0.0210, mod: 0.0118, hard: 0.0125
[INFO]: 2020-03-01 17:54:55,346 test_iter 50000 3d cyclist --> easy: 0.0141, mod: 0.0092, hard: 0.0095
